stackslib/src/chainstate/stacks/boot/pox_4_tests.rs

```diff
@@ -1698,7 +1745,195 @@ fn delegate_stack_stx_signer_key() {
     .expect_tuple();
 
     let state_signer_key = stacking_state.get("signer-key").unwrap();
-    assert_eq!(state_signer_key.to_string(), signer_key_val.to_string());
+    assert_eq!(
+        state_signer_key.to_string(),
+        format!("0x{}", signer_public_key.to_hex())
+    );
```

Comment:
I think it's better to compare the values directly instead of formatting to string.